lib/src/lightclient.rs

```diff
@@ -1173,156 +1021,23 @@ impl LightClient {
-            self.config.chain.hrp_unified_full_viewing_key() => todo!(),
-        )
-    }
-
-    /// Import a new transparent private key
-    pub async fn do_import_tk(&self, sk: String) -> Result<JsonValue, String> {
-        if !self.wallet.is_unlocked_for_spending().await {
-            error!("Wallet is locked");
-            return Err("Wallet is locked".to_string());
-        }
-
-        let address = self.wallet.add_imported_tk(sk).await;
-        if address.starts_with("Error") {
-            let e = address;
-            error!("{}", e);
-            return Err(e);
-        }
-
-        self.do_save().await?;
-        Ok(array![address])
-    }
-
-    /// Import a new orchard private key
-    pub async fn do_import_orchard_secret_key(
-        &self,
-        key: String,
-        birthday: u64,
-    ) -> Result<JsonValue, String> {
-        if !self.wallet.is_unlocked_for_spending().await {
-            error!("Wallet is locked");
-            return Err("Wallet is locked".to_string());
-        }
-
-        let new_address = {
-            let addr = self
-                .wallet
-                .add_imported_orchard_spending_key(key, birthday)
-                .await;
-            if addr.starts_with("Error") {
-                let e = addr;
-                error!("{}", e);
-                return Err(e);
-            }
-
-            addr
-        };
-
-        self.do_save().await?;
-
-        Ok(array![new_address])
-    }
-
-    /// Import a new z-address private key
-    pub async fn do_import_sapling_spend_key(
-        &self,
-        sk: String,
-        birthday: u64,
-    ) -> Result<JsonValue, String> {
-        if !self.wallet.is_unlocked_for_spending().await {
-            error!("Wallet is locked");
-            return Err("Wallet is locked".to_string());
-        }
-
-        let new_address = {
-            let addr = self.wallet.add_imported_sapling_extsk(sk, birthday).await;
-            if addr.starts_with("Error") {
-                let e = addr;
-                error!("{}", e);
-                return Err(e);
-            }
-
-            addr
-        };
-
-        self.do_save().await?;
-
-        Ok(array![new_address])
-    }
-
-    /// Import a new viewing key
-    pub async fn do_import_sapling_full_view_key(
-        &self,
-        vk: String,
-        birthday: u64,
-    ) -> Result<JsonValue, String> {
-        if !self.wallet.is_unlocked_for_spending().await {
-            error!("Wallet is locked");
-            return Err("Wallet is locked".to_string());
-        }
-
-        let new_address = {
-            let addr = self.wallet.add_imported_sapling_extfvk(vk, birthday).await;
-            if addr.starts_with("Error") {
-                let e = addr;
-                error!("{}", e);
-                return Err(e);
-            }
-
-            addr
-        };
+        let new_address = self
+            .wallet
+            .unified_spend_auth()
```

Comment:
A `unified_spend_auth` is created in this fluent expression.   Shouldn't it be in charge exposing `new_addresses` via another call?

We definitely have the case of wanting to generate new addresses, and then not use them right away, why not use that same method of exposing previously created addresses, that's done in those cases?   Instead of returning the new UA here?

Reply:
Returning the new UA here is intended purely for convenience. I don't mind removing it.